process/main.py

```diff
@@ -524,49 +531,93 @@ def append_input(self):
             mfile_file.write("***********************************************")
             mfile_file.writelines(input_lines)
 
-    def validate_input(self):
-        """Checks the input IN.DAT file for any obsolete variables in the OBS_VARS dict contained
-        within obsolete_variables.py.
-        Then will print out what the used obsolete variables are (if any) before continuing the proces run.
+    def validate_input(self, replace_obsolete=False):
+        """
+        Checks the input IN.DAT file for any obsolete variables in the OBS_VARS dict contained
+        within obsolete_variables.py. If obsolete variables are found, and if `replace_obsolete`
+        is set to True, they are either removed or replaced by their updated names as specified
+        in the OBS_VARS dictionary.
+
+        Parameters:
+            replace_obsolete (bool): If True, modifies the IN.DAT file to replace or comment out
+                                    obsolete variables. If False, only reports obsolete variables.
         """
 
         obsolete_variables = ov.OBS_VARS
         obsolete_vars_help_message = ov.OBS_VARS_HELP
 
         filename = self.input_file
-
         variables_in_in_dat = []
+        modified_lines = []
+        changes_made = []  # To store details of the changes
+
         with open(filename, "r") as file:
             for line in file:
-                if line[0] == "*" or "=" not in line:
+                # Skip comment lines or lines without an assignment
+                if line.startswith("*") or "=" not in line:
+                    modified_lines.append(line)
                     continue
 
+                # Extract the variable name before the separator
+                variable_name = line.strip().split(" ", 1)[0]
+                variables_in_in_dat.append(variable_name)
+
+                # Check if the variable is obsolete and needs replacing
+                if variable_name in obsolete_variables:
+                    replacement = obsolete_variables.get(variable_name)
+
+                    if replace_obsolete:
+                        # Prepare replacement or removal
+                        if replacement is None:
+                            # If no replacement is defined, comment out the line
+                            modified_lines.append(f"* Obsolete: {line}")
+                            changes_made.append(
+                                f"Commented out obsolete variable: {variable_name}"
+                            )
+                        else:
```

Comment:
I hadn't considered that there were lists of replacements. Not sure there is an ideal solution there... I don't really like the idea of writing the new variables with "wrong" values (for ex: the original value of replaced variable). I could just write them without values but this will "break" input file.

Reply:
Added error message for when there is a list of replacements.

Reply:
I agree with this solution ... if we cant fix it automatically, error